src/main/kotlin/com/cognifide/gradle/aem/base/vlt/CheckoutConfig.kt

```diff
@@ -0,0 +1,52 @@
+package com.cognifide.gradle.aem.base.vlt
+
+import com.cognifide.gradle.aem.api.AemConfig
+import com.cognifide.gradle.aem.instance.Instance
+import com.cognifide.gradle.aem.internal.file.FileOperations
+import org.gradle.api.Project
+import org.gradle.api.logging.Logger
+
+open class CheckoutConfig (private val project: Project, private val config: AemConfig) {
+
+    private val logger: Logger = project.logger
+
+    fun determineInstance(instanceConfig: String?): Instance {
+        if (!instanceConfig.isNullOrBlank()) {
+            val cmdInstance = config.parseInstance(instanceConfig!!)
+
+            logger.info("Using instance specified by command line parameter: $cmdInstance")
+            return cmdInstance
+        }
+
+        val namedInstance = Instance.filter(project, config.instanceName).firstOrNull()
+        if (namedInstance != null) {
+            logger.info("Using first instance matching filter '${config.instanceName}': $namedInstance")
```

Comment:
for me it does not make sense, because even you extracted it, instanceName is being determined by shared prop `aem.instance.name` but filter is being determined by `aem.download....` 

Instead of making it better, I think it is even more not consistent so that now I am rather more ok with your initial approach - reusing `aem.checkout...` properties in download. 

I dislike subconfig `CheckoutConfig`. Instead of doing more compicated structure, maybe we should stay with current. In next 6.x.x version, we will allow to specify all required properties directly in tasks, not in shared `AemConfig` but as I stated earlier, it will required major plugin redesign/refactoring and I am planning it for 6.x serie